src/macrofier/engine.mjs

```diff
@@ -1109,9 +1109,12 @@ function insertMethodMacros(template, methodObj, json, templates, examples = {})
     .replace(/\$\{method\.params\.array\}/g, JSON.stringify(methodObj.params.map(p => p.name)))
     .replace(/\$\{method\.params\.count}/g, methodObj.params ? methodObj.params.length : 0)
     .replace(/\$\{if\.params\}(.*?)\$\{end\.if\.params\}/gms, method.params.length ? '$1' : '')
-    .replace(/\$\{if\.params.empty\}(.*?)\$\{end\.if\.params.empty\}/gms, method.params.length === 0 ? '$1' : '')
+    .replace(/\$\{if\.result\}(.*?)\$\{end\.if\.result\}/gms, resultType ? '$1' : '')
+    .replace(/\$\{if\.params\.empty\}(.*?)\$\{end\.if\.params\.empty\}/gms, method.params.length === 0 ? '$1' : '')
+    .replace(/\$\{if\.signature\.empty\}(.*?)\$\{end\.if\.signature\.empty\}/gms, (method.params.length === 0 && resultType === '') ? '$1' : '')
```

Comment:
How could a method have no signature?

Reply:
if result of a method is, name:success and type:const/boolean. 
i.e, Discovery.signOut. Here result name is success and type is boolean, we don't want to consider this in function signature. similarly for type as const
if a method does not have any parameters and result as success + boolean/const:null. In such case the method signature become empty. C language we need to keep 'void' keyword for a function with out any parameter. Hence added signature.empty check to make sure appending this kind of function with 'void' keyword.

Reply:
ah okay. yeah... that's a tough big of logic to name, but I get why you called it `signature.empty`